src/Departments/Departments.js

```diff
@@ -12,33 +12,36 @@ class Departments extends Component {
   }
 
   componentDidMount () {
-    const getDepartaments = async () => {
-      const data = await API.getDepartaments()
-      if (data.statusText === "OK") {
-        this.setState({
-          departments: data.data
-        })
-      } else {
-        this.setState({
-          requestError: data
-        })
+      const getDepartaments = async () => {
+        try {
+          const data = await API.getDepartaments()
+          if (data.status === 200) {
+            this.setState({
+              departments: data.data
+            })
+          }
+        } catch (error) {
+          this.setState({
+            requestError: error.message
+          })
+        }
       }
-    };
-
-    getDepartaments();
+      getDepartaments()
   }
 
-  render() {
-    if (this.state.requestError) {
-      return <div>{this.state.requestError}</div>
-    }
-    return (
-      this.state.departments.map(({ id, name }) => {
-        return <div key={id}>
-          <Link to={`/departments/${id}`}>{name}</Link>
-        </div>
-      })
+  render () {
+    const {requestError, departments} = this.state
+
+    const dataToRender = departments.map(({id, name}) =>
+      <li key={id}>
+        <Link to={`/departments/${id}`}>{`${name}`}</Link>
+      </li>
     )
+
+    if (requestError) {
+      return <h1>{requestError}</h1>
+    }
```

Comment:
error handling or edge case scenarios always come first
no need to define the render function body if there is an error already